azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,494 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+import os
+from typing import List, Optional, Tuple
+
+from azure.core.paging import PageIterator
+from azure.core.exceptions import ResourceNotFoundError
+from knack.log import get_logger
+from rich.console import Console
+import yaml
+
+from azext_edge.edge.providers.orchestration.common import CUSTOM_LOCATIONS_API_VERSION
+from azext_edge.edge.providers.orchestration.resources.instances import Instances
+from azext_edge.edge.util.file_operations import read_file_content, validate_file_extension
+from azext_edge.edge.util.queryable import Queryable
+from azext_edge.edge.util.az_client import (
+    parse_resource_id,
+    get_keyvault_client,
+    get_ssc_mgmt_client,
+    wait_for_terminal_state,
+)
+
+logger = get_logger(__name__)
+
+console = Console()
+
+OPCUA_SPC_NAME = "opc-ua-connector"
+OPCUA_TRUST_LIST_SECRET_SYNC_NAME = "aio-opc-ua-broker-trust-list"
+OPCUA_ISSUER_LIST_SECRET_SYNC_NAME = "aio-opc-ua-broker-issuer-list"
+OPCUA_CLIENT_CERT_SECRET_SYNC_NAME = "aio-opc-ua-broker-client-certificate"
+SERVICE_ACCOUNT_NAME = "aio-ssc-sa"
+
+
+class OpcUACerts(Queryable):
+
+    def __init__(self, cmd):
+        super().__init__(cmd=cmd)
+        self.instances = Instances(self.cmd)
+        self.ssc_mgmt_client = get_ssc_mgmt_client(
+            subscription_id=self.default_subscription_id,
+        )
+        self
+
+    def trust_add(self, instance_name: str, resource_group: str, file: str, secret_name: Optional[str] = None) -> dict:
+        cl_resources = self._get_cl_resources(instance_name=instance_name, resource_group=resource_group)
+        secretsync_spc = self._find_existing_spc(instance_name=instance_name, cl_resources=cl_resources)
+        if not secretsync_spc:
+            return
+
+        # get properties from default spc
+        spc_properties = secretsync_spc.get("properties", {})
+        spc_keyvault_name = spc_properties.get("keyvaultName", "")
+        spc_tenant_id = spc_properties.get("tenantId", "")
```

Comment:
will things still work with no tenant id/client id/etc?

Reply:
nothing would work, if there is no default spc then we would not come to this stage, if default spc exists then we for sure have the tenant id/client id and keyvault name since they are required to provided in the secret sync enable step